File: go/scrabble-score/scrabble_score.go
package scrabble

import "unicode"

func Score(word string) (v int) {
	for _, r := range word {
		switch unicode.ToUpper(r) {
		case 'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T':
			v += 1
		case 'D', 'G':
			v += 2
		case 'B', 'C', 'M', 'P':
			v += 3
		case 'F', 'H', 'V', 'W', 'Y':
			v += 4
		case 'K':
			v += 5
		case 'J', 'X':
			v += 8
		case 'Q', 'Z':
			v += 10
		}
	}
	return
}

File: dart/scrabble-score/lib/scrabble_score.dart
// Put your code here
Map<String, int> scores = {
  'A': 1,
  'E': 1,
  'I': 1,
  'O': 1,
  'U': 1,
  'L': 1,
  'N': 1,
  'R': 1,
  'S': 1,
  'T': 1,
  'D': 2,
  'G': 2,
  'B': 3,
  'C': 3,
  'M': 3,
  'P': 3,
  'F': 4,
  'H': 4,
  'V': 4,
  'W': 4,
  'Y': 4,
  'K': 5,
  'J': 8,
  'X': 8,
  'Q': 10,
  'Z': 10,
};
int score(String word) {
  int sum = 0;
  for (var rune in word.runes) {
    String char = String.fromCharCode(rune).toUpperCase();
    if (scores.containsKey(char)) {
      sum += scores[char]?.toInt() ?? 0;
    }
  }
  return sum;
}
